modules/components/org.wso2.carbon.appfactory.s4.integration/src/main/java/org/wso2/carbon/appfactory/s4/integration/DomainMappingManagementService.java

```diff
@@ -137,6 +140,33 @@ public void addSubscriptionDomain(String stage, String domain, String appKey, St
     }
 
     /**
+     * Add default production url, if its failed during app creation time
+     *
+     * @param appKey    application key
+     * @param version   version of the application
+     * @throws AppFactoryException
+     */
+    public void addDefaultProdUrl(String appKey, String version) throws AppFactoryException {
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        String defaultHostName = ServiceReferenceHolder.getInstance().getAppFactoryConfiguration().getFirstProperty(
+                AppFactoryConstants.DOMAIN_NAME);
+        String defaultUrl = DomainMappingUtils.generateDefaultProdUrl(
+                appKey + AppFactoryConstants.MINUS + (new Random()).nextInt(1000),
+                tenantDomain, defaultHostName);
+        try {
+            addSubscriptionDomain(ServiceReferenceHolder.getInstance().getAppFactoryConfiguration().getFirstProperty(
+                    AppFactoryConstants.FINE_GRAINED_DOMAIN_MAPPING_ALLOWED_STAGE), defaultUrl, appKey, version, false);
+        } catch (DomainMappingVerificationException e) {
+            log.error(String.format(DomainMappingUtils.AF_CUSTOM_URL_NOT_VERIFIED, defaultUrl), e);
+            // update the custom url values of the rxt.
+            DomainMappingUtils.updateCustomUrlMetadata(appKey, defaultUrl,
```

Comment:
This seems wrong to me. Here we are trying to update the default production url. Therefore if an error occured we dont need to update the custom url metadata. Just log the error and throw it.